src/challenges/challengeOne/5.Modal/Modal.test.tsx

```diff
@@ -1,8 +1,57 @@
-// import React from "react";
-// import { render, screen } from "@testing-library/react";
-// import userEvent from "@testing-library/user-event";
-// import { Modal } from "./Modal";
-
-// describe("Modal", () => {
-//   it("renders...", () => {});
-// });
+import React from "react";
+import { render, screen } from "@testing-library/react";
+import userEvent from "@testing-library/user-event";
+import { Modal, ModalProps } from "./Modal";
+
+describe("Modal", () => {
+  const defaultProps: ModalProps = {
+    isOpen: true,
+    onClose: jest.fn(),
+    onConfirm: jest.fn(),
+    headerText: "Modal Header",
+    children: "Modal Content",
+  };
+
+  const renderComponent = (additionalProps?: Partial<ModalProps>) => {
+    const props: ModalProps = { ...defaultProps, ...additionalProps };
+    render(<Modal {...props} />);
+  };
+
+  it("renders the modal header and content correctly", () => {
+    renderComponent();
+
+    const header = screen.getByText("Modal Header");
+    const content = screen.getByText("Modal Content");
```

Comment:
Again I would say that this abstraction is unnecessary, ChatGPT will do this, but until you're using the variable multiple times, it's cleaner to just do 

`expect(screen.getByText("Modal Header")).toBeInTheDocument();`